From 424948e844ab296b4817dd019b11ed618d0233b6 Mon Sep 17 00:00:00 2001
From: Shilin Victor <chrono.monochrome@gmail.com>
Date: Mon, 1 Feb 2016 13:08:21 +0700
Subject: [PATCH 03/13] 
 0003-libhwui-Set-fall-back-layer-size-for-zero-size-reque.patch

Change-Id: I5e6f915bddb8c3e389f9dfe0362182d93a6661a3
---
 libs/hwui/LayerRenderer.cpp | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/libs/hwui/LayerRenderer.cpp b/libs/hwui/LayerRenderer.cpp
index 103c843..3d4c76f 100644
--- a/libs/hwui/LayerRenderer.cpp
+++ b/libs/hwui/LayerRenderer.cpp
@@ -195,6 +195,12 @@ Layer* LayerRenderer::createRenderLayer(RenderState& renderState, uint32_t width
         return NULL;
     }
 
+    if (width == 0 && height == 0) {
+        ALOGW("Trying to obtain a zero-size layer, setting fall-back size 512x768");
+        width = 512;
+        height = 768;
+    }
+
     caches.activeTexture(0);
     Layer* layer = caches.layerCache.get(renderState, width, height);
     if (!layer) {
-- 
2.5.0

